### ISSUE TITLE
`KeyError: 'layers'` error when using docker_pull with a digest

### ISSUE THREAD

**Issue by razvanm** [bazelbuild/rules_docker#167]
`docker_pull` works fine with `tag` but when I indicate a `digest` I hit a `KeyError: 'layers'` error. Any idea what I'm doing wrong?

From me `WORKSPACE`:

```
docker_pull(
    name = "docker_mysql_5_6",
    registry = "index.docker.io",
    repository = "library/mysql",
    tag = "5.6",
    digest = "sha256:4520321978cf1ddce0827aa406f934bb8a7aa766260b1d530ff36de0f7ca484a",
)
```

The  error:

```
$ bazel build @docker_mysql_5_6//...
ERROR: Pull command failed: Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "[REDACTED]/6f05adc21829d12463be186661011f76/external/puller/file/puller.par/__main__.py", line 83, in <module>
  File "[REDACTED]/6f05adc21829d12463be186661011f76/external/puller/file/puller.par/__main__.py", line 72, in main
  File "[REDACTED]/6f05adc21829d12463be186661011f76/external/puller/file/puller.par/containerregistry/client/v2_2/save_.py", line 168, in fast
  File "[REDACTED]/6f05adc21829d12463be186661011f76/external/puller/file/puller.par/containerregistry/client/v2_2/docker_image_.py", line 48, in fs_layers
KeyError: 'layers'
 ([REDACTED]/6f05adc21829d12463be186661011f76/external/puller/file/puller.par --directory [REDACTED]/6f05adc21829d12463be186661011f76/external/docker_mysql_5_6/image --name index.docker.io/library/mysql@sha256:4520321978cf1ddce0827aa406f934bb8a7aa766260b1d530ff36de0f7ca484a)
INFO: Elapsed time: 0.886s
FAILED: Build did NOT complete successfully (0 packages loaded)
$ 
```

Proof that the digest exists:

```
$ docker pull mysql:5.6@sha256:4520321978cf1ddce0827aa406f934bb8a7aa766260b1d530ff36de0f7ca484a
sha256:4520321978cf1ddce0827aa406f934bb8a7aa766260b1d530ff36de0f7ca484a: Pulling from library/mysql
Digest: sha256:4520321978cf1ddce0827aa406f934bb8a7aa766260b1d530ff36de0f7ca484a
Status: Image is up to date for mysql@sha256:4520321978cf1ddce0827aa406f934bb8a7aa766260b1d530ff36de0f7ca484a
$ 
```

**Comment by mattmoor** [comment]
Hey @razvanm, been a while.  I hope things are going well for you :)

Related-to / caused-by [this](https://integratedcode.us/2017/09/13/dockerhub-official-images-go-multi-platform/).

Essentially the digest you are passing isn't the image digest, but the digest of a collection of images, which are stitched together into a "fat" image.
```json
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 2616,
         "digest": "sha256:80efa2e28c78995d504dbda93c83cb279d0b1b0eecf77bcf3cb24c028e9ecc71",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      }
   ]
}
```

You can work around this by switching your digest to `sha256:80efa2e28c78995d504dbda93c83cb279d0b1b0eecf77bcf3cb24c028e9ecc71`.  It also works if you just use a tag, since finding this hash is non-trivial.

I'll use this to track accepting manifest list digests in the tool.

**Comment by mattmoor** [comment]
@jonjohnsonjr I'm going to kick this over to you since you added manifest list to the client.  If what's needed is unclear, grab me and we can chat f2f.

**Comment by jonjohnsonjr** [comment]
This should be fixed by #321.
